### PR TITLE
SONARIAC-687 Extract common code from CfnLintImporter

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: iac-common/src/main/java/org/sonar/iac/common/reports/AbstractJsonReportImporter.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      int failedToSaveIssues = saveIssues(issuesJson);
+      if (failedToSaveIssues > 0) {
+        addWarning(reportFile.getPath(), issuesJson.size(), failedToSaveIssues);
+      }
+    });
+  }
+
+  protected abstract void saveIssue(JSONObject issueJson);
+
+  protected Optional<JSONArray> parseJson(File reportFile) {
```
Comment (jonas-wielage-sonarsource) [inline iac-common/src/main/java/org/sonar/iac/common/reports/AbstractJsonReportImporter.java:71]: Looks like a method the `JSONParser` should contain itself.

File: iac-common/src/main/java/org/sonar/iac/common/reports/AbstractJsonReportImporter.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      String message = String.format("could not parse file as JSON %s", reportFile.getPath());
+      logWarning(message);
+    } catch (RuntimeException e) {
+      String message = String.format("file is expected to contain a JSON array but didn't %s", reportFile.getPath());
+      logWarning(message);
+    }
+    return Optional.ofNullable(issuesJson);
+  }
+
+  protected int saveIssues(JSONArray issuesJson) {
```
Comment (jonas-wielage-sonarsource) [inline iac-common/src/main/java/org/sonar/iac/common/reports/AbstractJsonReportImporter.java:88]: I would order it under the other `saveIssues(...) `method

**Review by jonas-wielage-sonarsource — APPROVED** [review]
LGTM!
But I'm wondering why the code coverage is so low? Shouldn't there be an IT that uses this?

**Comment by jonas-wielage-sonarsource** [comment]
And another question. Do you know why we don't use a JSON Parser framework like Jackson?

File: iac-common/src/main/java/org/sonar/iac/common/reports/AbstractJsonReportImporter.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+public abstract class AbstractJsonReportImporter {
+  private static final Logger LOG = Loggers.get(AbstractJsonReportImporter.class);
+  private static final JSONParser jsonParser = new JSONParser();
+
+  protected final SensorContext context;
+  private final AnalysisWarningsWrapper analysisWarnings;
+  private final String warningPrefix;
+  private Set<String> unresolvedPaths;
+
+  protected AbstractJsonReportImporter(SensorContext context, AnalysisWarningsWrapper analysisWarnings, String warningPrefix) {
```
Comment (jonas-wielage-sonarsource) [inline iac-common/src/main/java/org/sonar/iac/common/reports/AbstractJsonReportImporter.java:47]: I'm not sure about this, as i don't have that much experience with our naming convention.
Shouldn't we drop the `Abstract` in the class name?

File: iac-common/src/main/java/org/sonar/iac/common/reports/AbstractJsonReportImporter.java (hunk 1 above)
Comment (mstachniuk) [reply iac-common/src/main/java/org/sonar/iac/common/reports/AbstractJsonReportImporter.java:69]: The `JSONParser` contains already the `parse` method. Here in this method, an error handling is encapsulated, so the main flow in `importReport` is easier to read & understand.

File: iac-common/src/main/java/org/sonar/iac/common/reports/AbstractJsonReportImporter.java (hunk 3 above)
Comment (mstachniuk) [reply iac-common/src/main/java/org/sonar/iac/common/reports/AbstractJsonReportImporter.java:47]: We don't have a naming convention for that and for me, it's ok to use `Abstract` in the class name.

**Comment by jonas-wielage-sonarsource** [comment]
Thanks for the clarification about the coverage, makes sense that the coverage is only calculated module based.

**Pull request merged**